build/Microsoft.DotNet.Cli.Test.targets

```diff
@@ -0,0 +1,157 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Layout" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="test/ProjectsToTest.props" />
+  <Import Project="test/TestPackageProjects.props" />
+
+  <Target Name="PrepareTests"
+          DependsOnTargets="Prepare">
+
+    <PropertyGroup>
+      <TestPackageBuildVersionSuffix>$(CommitCount)</TestPackageBuildVersionSuffix>
+      <TestOutputDir>$(BaseOutputDirectory)/tests</TestOutputDir>
+      <TestPackagesDir>$(TestOutputDir)/packages</TestPackagesDir>
+      <TestArtifactsDir>$(TestOutputDir)/artifacts</TestArtifactsDir>
+      <TestResultXmlDir>$(TestOutputDir)/results</TestResultXmlDir>
+      <TestDirectory>$(RepoRoot)/test</TestDirectory>
+      <DotnetUnderTest>$(DotnetStage2)</DotnetUnderTest>
+    </PropertyGroup>    
+
+  </Target>
+
+  <Target Name="Test" 
+          DependsOnTargets="PrepareTests;
+                            SetupTests;
+                            RestoreTests;
+                            BuildTests;
+                            RunTests;" />
+
+  <Target Name="SetupTests"
+          DependsOnTargets="SetupTestProjects;" />
+
+  <Target Name="SetupTestProjects"
+          DependsOnTargets="SetupTestPackageProjectData;
+                            RestoreTestAssetProjects;
+                            RestoreDesktopTestAssetProjects;
+                            BuildTestAssetProjects;
+                            BuildDesktopTestAssetProjects;" />
+
+  <Target Name="SetupTestPackages" 
+          DependsOnTargets="RestoreTestAssetPackages;
+                            SetupTestPackageProjectData"
+          Outputs="%(TestPackageProject.Identity)">
```

Comment:
I get the outputs are for target batching but agreed matching inputs would be good for incrementality here.